### PR TITLE
feat(universal-chart): add rollout lifecycle controls

### LINKED ISSUE #186: Add structured Deployment rollout and lifecycle controls
## Problem

Applications can currently reach some container lifecycle fields through `extraContainerProps`, but common rollout controls are not first-class. Apps that need a deliberate Recreate strategy, preStop drain, minimum readiness window, or progress deadline must rely on loosely structured overrides or another chart.

## Proposed behavior

Add values for:

- Deployment strategy, including RollingUpdate parameters and Recreate;
- `minReadySeconds`;
- `progressDeadlineSeconds`;
- container lifecycle hooks, especially `preStop`; and
- documented interaction with readiness probes and `terminationGracePeriodSeconds`.

## Acceptance criteria

- RollingUpdate and Recreate render without conflicting fields.
- Lifecycle hooks are schema-validated and work for the main container.
- Tests cover defaults, custom rolling parameters, Recreate, and preStop.
- README guidance explains mixed-version compatibility and the availability cost of Recreate.

### PR DESCRIPTION
**Pull request by dannysauer**
## Summary

- add schema-backed RollingUpdate and Recreate strategies, minReadySeconds, and progressDeadlineSeconds
- add validated postStart and preStop handlers for the main container while preserving legacy extraContainerProps.lifecycle configurations
- reject rollout settings that cannot make progress, including an implicit 600-second deadline that is not greater than minReadySeconds
- document mixed-version rollouts, Recreate downtime, readiness timing, lifecycle compatibility, and termination grace periods

All new values are unset by default, so existing rendered Deployments do not change.

## Verification

- `make test` (244 passed)
- `pre-commit run --all-files`
- focused rollout and universal-chart golden tests (52 passed)

Closes #186

### CODE REVIEW

**Comment by dannysauer** [comment]
@codex review

File: charts/universal-chart/values.yaml [hunk 1]
```diff
@@ -673,6 +730,82 @@ readinessProbe:
...
+#         path:
+#           type: string
+#           pattern: ^/.*
+#         port:
+#           anyOf:
+#             - type: integer
+#               minimum: 1
+#               maximum: 65535
+#             - type: string
+#               minLength: 1
```
[the remark this answers is not in the document]
Comment (dannysauer) [reply charts/universal-chart/values.yaml:785]: Fixed in 7afddef. String ports now follow Kubernetes' port-name rules: at most 15 characters, lowercase alphanumerics with single internal hyphens, and at least one letter. The tests cover each rejected form and valid boundary cases.

File: charts/universal-chart/templates/deployment.yaml [hunk 2]
```diff
@@ -1,3 +1,22 @@
...
+{{- $progressDeadline = int .Values.deployment.progressDeadlineSeconds }}
+{{- end }}
+{{- if and (ne .Values.deployment.minReadySeconds nil) (le $progressDeadline (int .Values.deployment.minReadySeconds)) }}
+{{- fail "deployment.progressDeadlineSeconds must be greater than deployment.minReadySeconds." }}
+{{- end }}
+{{- $strategy := .Values.deployment.strategy | default dict }}
+{{- if eq (get $strategy "type") "RollingUpdate" }}
+{{- $rollingUpdate := get $strategy "rollingUpdate" | default dict }}
+{{- $maxSurgeIsZero := and (hasKey $rollingUpdate "maxSurge") (or (eq (toString (get $rollingUpdate "maxSurge")) "0") (eq (toString (get $rollingUpdate "maxSurge")) "0%")) }}
+{{- $maxUnavailableIsZero := and (hasKey $rollingUpdate "maxUnavailable") (or (eq (toString (get $rollingUpdate "maxUnavailable")) "0") (eq (toString (get $rollingUpdate "maxUnavailable")) "0%")) }}
```
[the remark this answers is not in the document]
Comment (dannysauer) [reply charts/universal-chart/templates/deployment.yaml]: Fixed in 7afddef. Zero detection now handles any all-zero percentage, so the value 00% with another zero is rejected while 01% remains valid. I added regression tests for both cases.

File: charts/universal-chart/values.yaml [hunk 3]
```diff
@@ -673,6 +730,82 @@ readinessProbe:
...
+#     exec:
+#       type: object
+#       additionalProperties: false
+#       properties:
+#         command:
+#           type: array
+#           minItems: 1
+#           items:
+#             type: string
+#             minLength: 1
```
[the remark this answers is not in the document]
Comment (dannysauer) [reply charts/universal-chart/values.yaml]: Fixed in 7afddef. The schema still requires a nonempty command array but now allows empty argument strings. A regression test covers an executable followed by an empty positional argument.

**Comment by dannysauer** [comment]
@codex review
